pkg/services/auth/auth.go

```diff
@@ -67,8 +68,12 @@ type RotateCommand struct {
 
 // UserTokenService are used for generating and validating user tokens
 type UserTokenService interface {
-	CreateToken(ctx context.Context, user *user.User, clientIP net.IP, userAgent string) (*UserToken, error)
+	CreateToken(ctx context.Context, user *user.User, clientIP net.IP, userAgent string, extSession *ExternalSession) (*UserToken, error)
```

Comment:
Consider adding a command struct, this function now has a lot or parameters now

```
struct CreateTokenCommand {
  userID int64
  clientIP net.IP
  userAgent string
  extSession *ExternalSession
}
```

Reply:
Done, currently I put everything 1-1 to `CreateTokenCommand` and will replace `User *user.User` with `UserID int64` in a followup PR.